components/eamxx/src/share/field/field_layout.cpp

```diff
@@ -126,36 +166,38 @@ LayoutType get_layout_type (const std::vector<FieldTag>& field_tags) {
     std::vector<FieldTag> lev_tags = {LEV,ILEV};
     return ekat::contains(lev_tags,t);
   };
-  auto is_vec_tag = [](const FieldTag t) {
-    std::vector<FieldTag> vec_tags = {CMP,NGAS,SWBND,LWBND,SWGPT,ISCCPTAU,ISCCPPRS};
-    return ekat::contains(vec_tags,t);
+  auto is_cmp_tag = [](const FieldTag t) {
```

Comment:
Does this mean we are relying on Tensor layouts to be like "DYNAMIC" above, i.e. we don't trigger a `return` statement from the conditional?  If so we might want to change the comment to include "A Dynamics or Tensor layout"

Reply:
No, it can be a dynamics layout or a physics one. The if/else above takes care of eliminating EL/COL/GP tags, so that we are left with just vector/tensor tags and possibly LEV/ILEV/TL.